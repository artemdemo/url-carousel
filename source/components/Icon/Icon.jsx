import React from 'react';
import classnames from 'classnames';

import './Icon.less';

const Icon = (props) => {
    const { className, name, flat } = props;
    const containerClass = classnames(className, {
        icon: true,
        icon_flat: flat,
    });
    const iconClass = classnames({
        glyphicon: true,
        [`glyphicon-${name}`]: name,
    });

    return (
        <span className={containerClass}>
            <i className={iconClass} />
        </span>
    );
};

Icon.propTypes = {
    className: React.PropTypes.string,
    name: React.PropTypes.string,
    flat: React.PropTypes.bool,
};

export default Icon;
