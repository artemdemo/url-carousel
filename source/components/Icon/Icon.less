.icon {
    display: inline-block;
    margin-right: 5px;
}
    .icon_flat {
        display: inline;
        margin: 0;
    }
