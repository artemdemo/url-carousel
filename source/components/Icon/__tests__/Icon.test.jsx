import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('<Icon>', () => {
    it('Simple render', () => {
        const tree = renderer.create(
            <Icon name='ok' />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('With className', () => {
        const tree = renderer.create(
            <Icon name='remove' className='additional-class' />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('flat', () => {
        const tree = renderer.create(
            <Icon name='remove' flat />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
