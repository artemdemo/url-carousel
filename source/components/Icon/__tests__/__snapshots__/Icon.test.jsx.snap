// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> Simple render 1`] = `
<span
  className="icon"
>
  <i
    className="glyphicon glyphicon-ok"
  />
</span>
`;

exports[`<Icon> With className 1`] = `
<span
  className="additional-class icon"
>
  <i
    className="glyphicon glyphicon-remove"
  />
</span>
`;

exports[`<Icon> flat 1`] = `
<span
  className="icon icon_flat"
>
  <i
    className="glyphicon glyphicon-remove"
  />
</span>
`;
